### PR TITLE
Several fixes to improve the stability of E2E testing

### PR DESCRIPTION
**Pull request by ywk253100**
1. Support to customize the restic restore helper image
2. Use a seperated context when doing the clean up works
3. Wait a while before doing the the restore for aws to avoid #1799

Signed-off-by: Wenkai Yin(尹文开) <yinw@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: test/e2e/kibishii_tests.go [hunk 1]
```diff
@@ -144,6 +145,13 @@ func runKibishiiTests(client testClient, providerName, veleroCLI, veleroNamespac
 		return errors.Wrapf(err, "failed to delete namespace %s", kibishiiNamespace)
 	}
 
+	// the snapshots of AWS may be still in pending status when do the restore, wait for a while
+	// to avoid this https://github.com/vmware-tanzu/velero/issues/1799
+	// TODO remove this after https://github.com/vmware-tanzu/velero/issues/3533 is fixed
+	if providerName == "aws" && useVolumeSnapshots {
```
Comment (zubron) [inline test/e2e/kibishii_tests.go:151]: Would the other providers benefit from waiting here as well? I'm not sure how long vSphere/Azure snapshots take. Are you only seeing issues with AWS?

**Review by zubron — APPROVED** [review]
LGTM - thanks! I just have one question about waiting for volume snapshots. I'm going to go ahead and merge though, and we can follow up in another PR if you think we should wait for the snapshots on other providers :+1:

File: test/e2e/kibishii_tests.go (hunk 1 above)
Comment (sseago) [reply test/e2e/kibishii_tests.go:151]: @zubron I know from prior experience with the snapshot wait issue that it's definitely a concern for Azure and GCP as well. When we implemented a workaround for this in our konveyor forks for the plugins, the fix was needed for all 3.

Comment (zubron) [reply test/e2e/kibishii_tests.go:151]: Thanks for the info, @sseago. I'll open another issue to follow up on this and wait for all snapshots.

**Pull request merged**